cmd/maya-exporter/app/collector/zvol/metrics.go

```diff
@@ -65,224 +69,338 @@ type fields struct {
-		),
-
-		rebuildBytes: prometheus.NewGaugeVec(
-			prometheus.GaugeOpts{
-				Namespace: "openebs",
-				Name:      "rebuild_bytes",
-				Help:      "Rebuild bytes",
-			},
-			[]string{"vol", "pool"},
-		),
-
-		rebuildStatus: prometheus.NewGaugeVec(
-			prometheus.GaugeOpts{
-				Namespace: "openebs",
-				Name:      "rebuild_status",
-				Help:      `Status of rebuild on replica (0, 1, 2, 3, 4, 5, 6)= {"INIT", "DONE", "SNAP REBUILD INPROGRESS", "ACTIVE DATASET REBUILD INPROGRESS", "ERRORED", "FAILED", "UNKNOWN"}`,
-			},
-			[]string{"vol", "pool"},
-		),
-
-		rebuildDoneCount: prometheus.NewGaugeVec(
-			prometheus.GaugeOpts{
-				Namespace: "openebs",
-				Name:      "total_rebuild_done",
-				Help:      "Total no of rebuild done on replica",
-			},
-			[]string{"vol", "pool"},
-		),
-
-		rebuildFailedCount: prometheus.NewGaugeVec(
-			prometheus.GaugeOpts{
-				Namespace: "openebs",
-				Name:      "total_failed_rebuild",
-				Help:      "Total no of failed rebuilds on replica",
-			},
-			[]string{"vol", "pool"},
-		),
-
-		zfsCommandErrorCounter: prometheus.NewGauge(
-			prometheus.GaugeOpts{
-				Namespace: "openebs",
-				Name:      "zfs_command_error",
-				Help:      "Total no of zfs command errors",
-			},
-		),
-
-		zfsStatsParseErrorCounter: prometheus.NewGauge(
-			prometheus.GaugeOpts{
-				Namespace: "openebs",
-				Name:      "zfs_stats_parse_error_counter",
-				Help:      "Total no of zfs stats parse errors",
-			},
-		),
-
-		zfsStatsRejectRequestCounter: prometheus.NewGauge(
-			prometheus.GaugeOpts{
-				Namespace: "openebs",
-				Name:      "zfs_stats_reject_request_count",
-				Help:      "Total no of rejected requests of zfs stats",
-			},
-		),
-	}
+func newMetrics() *metrics {
+	return new(metrics)
+}
+
+func (m *metrics) withReadBytes() *metrics {
+	m.readBytes = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "openebs",
+			Name:      "total_read_bytes",
+			Help:      "Total read in bytes",
+		},
+		[]string{"vol", "pool"},
+	)
+	return m
+}
+
+func (m *metrics) withWriteBytes() *metrics {
+	m.writeBytes = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "openebs",
+			Name:      "total_write_bytes",
+			Help:      "Total write in bytes",
+		},
+		[]string{"vol", "pool"},
+	)
+	return m
+}
+
+func (m *metrics) withReadCount() *metrics {
+	m.readCount = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "openebs",
+			Name:      "total_read_count",
+			Help:      "Total read io count",
+		},
+		[]string{"vol", "pool"},
+	)
+	return m
+}
+
+func (m *metrics) withWriteCount() *metrics {
+	m.writeCount = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "openebs",
+			Name:      "total_write_count",
+			Help:      "Total write io count",
+		},
+		[]string{"vol", "pool"},
+	)
+	return m
+}
+
+func (m *metrics) withSyncCount() *metrics {
+	m.syncCount = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "openebs",
+			Name:      "sync_count",
```

Comment:
@utkarshmani1997 If it is possible can we make this metric name more descriptive like what sync_count (on replica). Ignore the comment if you already have thought about it and the way the metric name is, is the right way.